### PR TITLE
Remove travis in favor of CircleCI

### PR DESCRIPTION
**Pull request by saschagrunert**
Hey, do you think we're ready to go for CircleCI? It seems pretty stable to me right now.

What I did:
- Pull out travis configuration
- Update `git-validation` target to check against the merge-base per
  default
- Rename `TRAVIS` variable to `CI` (automatically set by CircleCI)
- Add Dockerfile-* modeline for vim

### CODE REVIEW

**Comment by haircommander** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@cevich and/or @edsantiago WDYT?

File: test/cgroups.bats [hunk 1]
```diff
@@ -40,7 +40,7 @@ function teardown() {
 }
 
 @test "conmon custom cgroup" {
-	if [[ "$TRAVIS" == "true" ]]; then
+	if [[ "$CI" == "true" ]]; then
```
Comment (edsantiago) [inline test/cgroups.bats]: Does `$CI` have the same semantics (i.e. set to "true") as `$TRAVIS`?

File: test/cgroups.bats [hunk 2]
```diff
@@ -40,7 +40,7 @@ function teardown() {
...
 		skip "travis container tests don't support systemd cgroups"
```
Comment (edsantiago) [inline test/cgroups.bats]: Is this still true? If so, should `travis` be changed to 'CircleCI`?

Same comments apply below (to ctr.bats)

File: test/cgroups.bats (hunk 1 above)
Comment (saschagrunert) [reply test/cgroups.bats]: Yes, that's completely the same. :+1:

File: test/cgroups.bats (hunk 2 above)
Comment (saschagrunert) [reply test/cgroups.bats]: Definitely, I will adapt it.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch to get in the latest CI fixes.

**Comment by haircommander** [comment]
Bumping this because travis is horribly slow on master atm @umohnani8 @mrunalp PTAL

**Comment by mrunalp** [comment]
I am in favor of moving to circle ci, however, why don't we see it actually running on all master PRs? See https://github.com/cri-o/cri-o/pull/2525 for exampe.

**Comment by saschagrunert** [comment]
> I am in favor of moving to circle ci, however, why don't we see it actually running on all master PRs? See #2525 for exampe.

It’s currently not enabled for all forked PRs, but we can enable it here under “Build forked Pull requests”: https://circleci.com/gh/cri-o/cri-o/edit#advanced-settings

:)

**Comment by mrunalp** [comment]
@saschagrunert done :) Thanks!

**Comment by saschagrunert** [comment]
Cool, thanks 🙏

**Comment by rhatdan** [comment]
@saschagrunert Interested in doing similar for Buildah and Container/storage?

**Comment by saschagrunert** [comment]
> @saschagrunert Interested in doing similar for Buildah and Container/storage?

Oh yeah if you want then I can follow up on that. :)

**Pull request merged**